test/conversations/Conversation.test.ts

```diff
@@ -18,182 +29,365 @@ describe('conversation', () => {
-      bob.address
-    )
+      expect(bobMessages).toHaveLength(2)
+    })
+
+    it('lists paginated messages', async () => {
+      const aliceConversation = await alice.conversations.newConversation(
+        bob.address
+      )
+
+      for (let i = 0; i < 10; i++) {
+        await aliceConversation.send('gm')
+      }
+      await sleep(100)
+
+      let numPages = 0
+      const messageIds = new Set<string>()
+      for await (const page of aliceConversation.messagesPaginated({
+        pageSize: 5,
+      })) {
+        numPages++
+        expect(page).toHaveLength(5)
+        for (const msg of page) {
+          expect(msg.content).toBe('gm')
+          messageIds.add(msg.id)
+        }
+      }
+      expect(numPages).toBe(2)
+      expect(messageIds.size).toBe(10)
 
-    for (let i = 0; i < 10; i++) {
+      // Test sorting
+      let lastMessage: DecodedMessage | undefined = undefined
+      for await (const page of aliceConversation.messagesPaginated({
+        direction: SortDirection.SORT_DIRECTION_DESCENDING,
+      })) {
+        for (const msg of page) {
+          if (lastMessage && lastMessage.sent) {
+            expect(msg.sent?.valueOf()).toBeLessThanOrEqual(
+              lastMessage.sent?.valueOf()
+            )
+          }
+          expect(msg).toBeInstanceOf(DecodedMessage)
+          lastMessage = msg
+        }
+      }
+    })
+
+    it('ignores failed decoding of messages', async () => {
+      const aliceConversation = await alice.conversations.newConversation(
+        bob.address
+      )
+
+      // This should be readable
       await aliceConversation.send('gm')
-    }
-    await sleep(100)
+      // This should not be readable
+      await alice.publishEnvelopes([
+        {
+          message: Uint8Array.from([1, 2, 3]),
+          contentTopic: buildDirectMessageTopic(alice.address, bob.address),
+        },
+      ])
+      await sleep(100)
 
-    let numPages = 0
-    const messageIds = new Set<string>()
-    for await (const page of aliceConversation.messagesPaginated({
-      pageSize: 5,
-    })) {
-      numPages++
-      expect(page).toHaveLength(5)
-      for (const msg of page) {
-        expect(msg.content).toBe('gm')
-        messageIds.add(msg.id)
+      let numMessages = 0
+      for await (const page of aliceConversation.messagesPaginated()) {
+        numMessages += page.length
       }
-    }
-    expect(numPages).toBe(2)
-    expect(messageIds.size).toBe(10)
-
-    // Test sorting
-    let lastMessage: MessageV1 | MessageV2 | undefined = undefined
-    for await (const page of aliceConversation.messagesPaginated({
-      direction: SortDirection.SORT_DIRECTION_DESCENDING,
-    })) {
-      for (const msg of page) {
-        if (lastMessage && lastMessage.sent) {
-          expect(msg.sent?.valueOf()).toBeLessThanOrEqual(
-            lastMessage.sent?.valueOf()
-          )
+      expect(numMessages).toBe(1)
+    })
+
+    it('works for messaging yourself', async () => {
```

Comment:
We had this test before in the SDK. I wonder if we should log a warning when someone does this

Reply:
Do you want to disallow this?

Reply:
I don't want to disallow it. But I suspect it's being done by mistake if any actual user is doing it. Or they are using the SDK as some sort of private storage library, which it really wasn't designed for.

Reply:
I see, I'm ok either way. I just don't find warnings a good way to guide users, it's more of a last resort to surface something if nothing better is available.